prime-router/src/test/kotlin/azure/BlobAccessTests.kt

```diff
@@ -52,4 +73,237 @@ class BlobAccessTests {
             assertThat(e).isNotNull()
         }
     }
+
+    @Test
+    fun `upload report`() {
+        val testUrl = "http://uploadreport"
+        val testFormat = Report.Format.CSV
+        val testBytes = "testbytes".toByteArray()
+
+        val one = Schema(name = "one", topic = Topic.TEST)
+        val metadata = Metadata(schema = one)
+        val report1 = Report(
+            one, listOf(listOf("1", "2"), listOf("3", "4")), source = TestSource,
+            bodyFormat = testFormat, metadata = metadata
+        )
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every {
+            BlobAccess.uploadBody(
+                report1.bodyFormat, testBytes, report1.name, null,
+                Event.EventAction.NONE
+            )
+        } returns
+            BlobAccess.BlobInfo(report1.bodyFormat, testUrl, BlobAccess.sha256Digest(testBytes))
+
+        val testBlob = BlobAccess()
+        val result = testBlob.uploadReport(report1, testBytes)
+
+        assertThat(result.format).isEqualTo(testFormat)
+        assertThat(result.blobUrl).isEqualTo(testUrl)
+        assertThat(result.digest).isEqualTo(BlobAccess.sha256Digest(testBytes))
+    }
+
+    @Test
+    fun `upload body`() {
+        val testUrl = "http://uploadbody"
+        val testFormat = Report.Format.CSV
+        val testName = "testblob"
+        val testBytes = "testbytes".toByteArray()
+        val testFolder = "testfolder"
+        val testEnv = "testenvvar"
+        val testEvents: List<Event.EventAction?> = listOf(
+            Event.EventAction.RECEIVE,
+            Event.EventAction.SEND,
+            Event.EventAction.BATCH,
+            Event.EventAction.PROCESS,
+            Event.EventAction.ROUTE,
+            Event.EventAction.TRANSLATE,
+            Event.EventAction.NONE,
+            Event.EventAction.CONVERT,
+            null
+        )
+
+        mockkClass(BlobAccess::class)
+        mockkObject(BlobAccess.Companion)
+        every { BlobAccess.Companion.getBlobConnection(testEnv) } returns "testconnection"
+        every { BlobAccess.Companion.uploadBlob(any(), testBytes) } returns testUrl
```

Comment:
Instead of using `any()` here and having them all return the same url, it might be nice to verify that each action actually results in the expected bloburl, by having a different `every { uploadBlob(<expected param for action>, testBytes) } returns <action-specific-url>` for each action. Makes this a bit more verbose, but I think it would help this feel like it's a productive test instead of a test purely for code coverage.

Reply:
I switched this test to capture the blobName and had the mock return it as the URL so that we could test the value is as expected.